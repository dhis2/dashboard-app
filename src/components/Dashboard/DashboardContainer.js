import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import classes from './styles/DashboardContainer.module.css'

const DashboardContainer = ({ children, covered }) => {
    return (
        <div
            className={cx(
                classes.container,
                'dashboard-scroll-container',
                covered && classes.covered
            )}
        >
            {children}
        </div>
    )
}

DashboardContainer.propTypes = {
    children: PropTypes.node,
    covered: PropTypes.bool,
}

export default DashboardContainer
