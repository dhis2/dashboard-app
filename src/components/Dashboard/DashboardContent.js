import React, { Fragment } from 'react';
import TitleBar from '../TitleBar/TitleBar';
import ItemGrid from '../ItemGrid/ItemGrid';

export const DashboardContent = props => (
    <Fragment>
        <TitleBar edit={props.editMode} />
        <ItemGrid edit={props.editMode} />
    </Fragment>
);

export default DashboardContent;
