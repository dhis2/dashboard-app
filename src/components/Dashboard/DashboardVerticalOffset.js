import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import {
    getControlBarHeight,
    MIN_ROW_COUNT,
} from '../ControlBar/controlBarDimensions';
import { sGetControlBarUserRows } from '../../reducers/controlBar';

const DashboardVerticalOffset = props => {
    const rows = props.editMode ? MIN_ROW_COUNT : props.userRows;
    const marginTop = getControlBarHeight(rows);

    return <div className="page-container-top-margin" style={{ marginTop }} />;
};

const mapStateToProps = state => ({
    userRows: sGetControlBarUserRows(state),
});

DashboardVerticalOffset.propTypes = {
    editMode: PropTypes.bool,
    userRows: PropTypes.number,
};

export default connect(mapStateToProps)(DashboardVerticalOffset);
