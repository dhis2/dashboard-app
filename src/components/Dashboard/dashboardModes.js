export const EDIT = 'edit';
export const NEW = 'new';
export const VIEW = 'view';
