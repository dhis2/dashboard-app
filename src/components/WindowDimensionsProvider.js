import React, { createContext, useContext, useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import debounce from 'lodash/debounce'

export const WindowDimensionsCtx = createContext(null)

const windowDims = () => ({
    height: window.innerHeight,
    width: window.innerWidth,
})

const WindowDimensionsProvider = ({ children }) => {
    const [dimensions, setDimensions] = useState(windowDims())

    useEffect(() => {
        const handleResize = debounce(() => {
            setDimensions(windowDims())
        }, 200)
        window.addEventListener('resize', handleResize)
        return () => {
            window.removeEventListener('resize', handleResize)
        }
    }, [])

    return (
        <WindowDimensionsCtx.Provider value={dimensions}>
            {children}
        </WindowDimensionsCtx.Provider>
    )
}

WindowDimensionsProvider.propTypes = {
    children: PropTypes.node.isRequired,
}

export default WindowDimensionsProvider

export const useWindowDimensions = () => {
    return useContext(WindowDimensionsCtx)
}
