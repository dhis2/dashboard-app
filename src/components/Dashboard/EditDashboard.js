import React, { Component } from 'react'
import { connect } from 'react-redux'
import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'

import { acSetEditDashboard } from '../../actions/editDashboard'
import { sGetSelectedId } from '../../reducers/selected'
import {
    sGetDashboardById,
    sGetDashboardItems,
    sDashboardsIsFetching,
} from '../../reducers/dashboards'
import DashboardVerticalOffset from './DashboardVerticalOffset'
import DashboardContent from './DashboardContent'
import EditBar from '../ControlBar/EditBar'
import NoContentMessage from '../../widgets/NoContentMessage'

export const Content = ({ updateAccess }) => {
    return updateAccess ? (
        <DashboardContent editMode={true} />
    ) : (
        <NoContentMessage text={i18n.t('No access')} />
    )
}

Content.propTypes = {
    updateAccess: PropTypes.bool,
}
export class EditDashboard extends Component {
    state = {
        initialized: false,
    }

    initEditDashboard = () => {
        if (this.props.dashboard) {
            this.setState({ initialized: true })
            this.props.setEditDashboard(this.props.dashboard, this.props.items)
        }
    }

    componentDidMount() {
        this.initEditDashboard()
    }

    componentDidUpdate() {
        if (!this.state.initialized) {
            this.initEditDashboard()
        }
    }

    getDashboardContent = () => {
        const contentNotReady =
            !this.props.dashboardsLoaded || this.props.id === null

        return (
            <div className="dashboard-wrapper">
                {contentNotReady ? null : (
                    <Content updateAccess={this.props.updateAccess} />
                )}
            </div>
        )
    }

    render() {
        return (
            <>
                <EditBar />
                <DashboardVerticalOffset editMode={true} />
                {this.getDashboardContent()}
            </>
        )
    }
}

EditDashboard.propTypes = {
    dashboard: PropTypes.object,
    dashboardsLoaded: PropTypes.bool,
    id: PropTypes.string,
    items: PropTypes.array,
    setEditDashboard: PropTypes.func,
    updateAccess: PropTypes.bool,
}

const mapStateToProps = state => {
    const id = sGetSelectedId(state)
    const dashboard = id ? sGetDashboardById(state, id) : null

    const updateAccess =
        dashboard && dashboard.access ? dashboard.access.update : false

    return {
        dashboard,
        id,
        updateAccess,
        items: sGetDashboardItems(state),
        dashboardsLoaded: !sDashboardsIsFetching(state),
    }
}

export default connect(mapStateToProps, {
    setEditDashboard: acSetEditDashboard,
})(EditDashboard)
