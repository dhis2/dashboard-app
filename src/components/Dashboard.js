import React, { useState, useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import isEmpty from 'lodash/isEmpty'
import i18n from '@dhis2/d2-i18n'
import {
    Layer,
    CenteredContent,
    CircularLoader,
    CssVariables,
    AlertStack,
    AlertBar,
} from '@dhis2/ui'
import { Redirect } from 'react-router-dom'
import { useD2 } from '@dhis2/app-runtime-adapter-d2'
import NoContentMessage from './NoContentMessage'
import { ViewDashboard, DashboardsBar } from '../pages/view/'
import { EditDashboard, NewDashboard } from '../pages/edit/'
import { PrintDashboard, PrintLayoutDashboard } from '../pages/print/'

import { acClearEditDashboard } from '../actions/editDashboard'
import {
    tSetSelectedDashboardById,
    acSetSelectedIsLoading,
} from '../actions/selected'
import {
    sGetDashboardById,
    sDashboardsIsFetching,
    sGetAllDashboards,
} from '../reducers/dashboards'
import { sGetSelectedId, NON_EXISTING_DASHBOARD_ID } from '../reducers/selected'
import {
    EDIT,
    NEW,
    VIEW,
    PRINT,
    PRINT_LAYOUT,
    isPrintMode,
    isEditMode,
} from '../modules/dashboardModes'

import { useWindowDimensions } from './WindowDimensionsProvider'
import { isSmallScreen } from '../modules/smallScreen'

const setHeaderbarVisibility = mode => {
    const header = document.getElementsByTagName('header')[0]
    if (isPrintMode(mode)) {
        header.classList.add('hidden')
    } else {
        header.classList.remove('hidden')
    }
}

const dashboardMap = {
    [VIEW]: <ViewDashboard />,
    [EDIT]: <EditDashboard />,
    [NEW]: <NewDashboard />,
    [PRINT]: <PrintDashboard />,
    [PRINT_LAYOUT]: <PrintLayoutDashboard />,
}

const Dashboard = ({
    id,
    name,
    mode,
    dashboardsLoaded,
    dashboardsIsEmpty,
    routeId,
    selectDashboard,
    clearEditDashboard,
    setIsLoading,
}) => {
    const { width } = useWindowDimensions()
    const [redirectUrl, setRedirectUrl] = useState(null)
    const [loadingMessage, setLoadingMessage] = useState(null)
    const { d2 } = useD2()
    const username = d2.currentUser.username

    useEffect(() => {
        setHeaderbarVisibility(mode)
    }, [mode])

    useEffect(() => {
        if (isSmallScreen(width) && isEditMode(mode)) {
            const redirectUrl = routeId ? `/${routeId}` : '/'
            setRedirectUrl(redirectUrl)
        } else {
            setRedirectUrl(null)
        }
    }, [mode])

    useEffect(() => {
        if (!isEditMode(mode)) {
            clearEditDashboard()
        }
    }, [mode])

    useEffect(() => {
        const prepareDashboard = async () => {
            setIsLoading(true)
            const alertTimeout = setTimeout(() => {
                if (name) {
                    setLoadingMessage(
                        i18n.t('Loading dashboard â€“ {{name}}', { name })
                    )
                } else {
                    setLoadingMessage(i18n.t('Loading dashboard'))
                }
            }, 500)

            await selectDashboard(routeId, mode, username)

            setIsLoading(false)
            clearTimeout(alertTimeout)
            setLoadingMessage(null)
        }
        if (dashboardsLoaded && !dashboardsIsEmpty && mode !== NEW) {
            prepareDashboard()
        }
    }, [dashboardsLoaded, dashboardsIsEmpty, routeId, mode])

    const getContent = () => {
        if (!dashboardsLoaded) {
            return (
                <Layer translucent>
                    <CenteredContent>
                        <CircularLoader />
                    </CenteredContent>
                </Layer>
            )
        }

        if (redirectUrl) {
            return <Redirect to={redirectUrl} />
        }

        if (mode === NEW) {
            return dashboardMap[mode]
        }

        if (dashboardsIsEmpty) {
            return (
                <>
                    <DashboardsBar />
                    <NoContentMessage
                        text={i18n.t(
                            'No dashboards found. Use the + button to create a new dashboard.'
                        )}
                    />
                </>
            )
        }

        if (id === NON_EXISTING_DASHBOARD_ID) {
            return (
                <>
                    <DashboardsBar />
                    <NoContentMessage
                        text={i18n.t('Requested dashboard not found')}
                    />
                </>
            )
        }

        if (id === null) {
            return (
                <Layer translucent>
                    <CenteredContent>
                        <CircularLoader />
                    </CenteredContent>
                </Layer>
            )
        }

        return dashboardMap[mode]
    }

    return (
        <>
            <CssVariables colors spacers />
            {getContent()}
            <AlertStack>
                {loadingMessage && (
                    <AlertBar
                        onHidden={() => setLoadingMessage(null)}
                        permanent
                    >
                        {loadingMessage}
                    </AlertBar>
                )}
            </AlertStack>
        </>
    )
}

Dashboard.propTypes = {
    clearEditDashboard: PropTypes.func,
    dashboardsIsEmpty: PropTypes.bool,
    dashboardsLoaded: PropTypes.bool,
    id: PropTypes.string,
    mode: PropTypes.string,
    name: PropTypes.string,
    routeId: PropTypes.string,
    selectDashboard: PropTypes.func,
    setIsLoading: PropTypes.func,
}

const mapStateToProps = (state, ownProps) => {
    const dashboards = sGetAllDashboards(state)
    // match is provided by the react-router-dom
    const routeId = ownProps.match?.params?.dashboardId || null

    return {
        dashboardsIsEmpty: isEmpty(dashboards),
        dashboardsLoaded: !sDashboardsIsFetching(state),
        id: sGetSelectedId(state),
        name: sGetDashboardById(state, routeId)?.displayName || null,
        routeId,
    }
}

export default connect(mapStateToProps, {
    selectDashboard: tSetSelectedDashboardById,
    clearEditDashboard: acClearEditDashboard,
    setIsLoading: acSetSelectedIsLoading,
})(Dashboard)
