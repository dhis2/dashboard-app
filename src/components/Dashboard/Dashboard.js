import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { tSelectDashboard } from '../../actions/dashboards'
import { sDashboardsIsFetching } from '../../reducers/dashboards'
import { EDIT, NEW } from './dashboardModes'
import ViewDashboard from './ViewDashboard'
import EditDashboard from './EditDashboard'
import NewDashboard from './NewDashboard'

class Dashboard extends Component {
    setDashboard = () => {
        if (this.props.dashboardsLoaded) {
            const id = this.props.match.params.dashboardId || null

            this.props.selectDashboard(id)

            this.scrollToTop()
        }
    }

    scrollToTop = () => {
        window.scrollTo(0, 0)
    }

    componentDidMount() {
        this.setDashboard()
    }

    componentDidUpdate() {
        this.setDashboard()
    }

    render() {
        switch (this.props.mode) {
            case EDIT:
                return <EditDashboard />
            case NEW:
                return <NewDashboard />
            default:
                return <ViewDashboard />
        }
    }
}

Dashboard.propTypes = {
    dashboardsLoaded: PropTypes.bool,
    match: PropTypes.object,
    mode: PropTypes.string,
    selectDashboard: PropTypes.func,
}

const mapStateToProps = state => {
    return { dashboardsLoaded: !sDashboardsIsFetching(state) }
}

export default connect(mapStateToProps, {
    selectDashboard: tSelectDashboard,
})(Dashboard)
