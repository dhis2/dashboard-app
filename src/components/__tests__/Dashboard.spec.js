import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { createMemoryHistory } from 'history'
import { useD2 } from '@dhis2/app-runtime-adapter-d2'
import { Router } from 'react-router-dom'
import thunk from 'redux-thunk'
import { useDataEngine } from '@dhis2/app-runtime'

import Dashboard from '../Dashboard'
import WindowDimensionsProvider from '../../components/WindowDimensionsProvider'
import {
    NEW,
    VIEW,
    EDIT,
    PRINT,
    PRINT_LAYOUT,
} from '../../modules/dashboardModes'
import { NON_EXISTING_DASHBOARD_ID } from '../../reducers/selected'
import { apiFetchDashboard } from '../../api/fetchDashboard'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

jest.mock('@dhis2/analytics', () => ({
    getDimensionById: jest.fn(),
    DIMENSION_ID_PERIOD: 'pe',
    DIMENSION_ID_ORGUNIT: 'ou',
}))
jest.mock('@dhis2/app-runtime-adapter-d2')
jest.mock('@dhis2/app-runtime')
jest.mock('../../api/fetchDashboard')
jest.mock('../../pages/print', () => ({
    __esModule: true,
    PrintDashboard: function MockComponent() {
        return <div>PrintDashboard</div>
    },
    PrintLayoutDashboard: function MockComponent() {
        return <div>PrintLayoutDashboard</div>
    },
}))

jest.mock('../../pages/edit', () => ({
    __esModule: true,
    EditDashboard: function MockComponent() {
        return <div>EditDashboard</div>
    },
    NewDashboard: function MockComponent() {
        return <div>NewDashboard</div>
    },
}))

jest.mock('../../pages/view', () => ({
    __esModule: true,
    ViewDashboard: function MockComponent() {
        return <div>ViewDashboard</div>
    },
    DashboardsBar: function MockComponent() {
        return <div>DashboardsBar</div>
    },
}))
/* eslint-disable react/prop-types */
jest.mock(
    '../../components/NoContentMessage',
    () =>
        function MockComponent({ text }) {
            return <div>{text}</div>
        }
)
/* eslint-enable react/prop-types */

jest.mock('@dhis2/ui', () => ({
    __esModule: true,
    Layer: function MockComponent() {
        return <div>Layer</div>
    },
    CenteredContent: function MockComponent() {
        return <div>CenteredContent</div>
    },
    CircularLoader: function MockComponent() {
        return <div>CircularLoader</div>
    },
    CssVariables: function MockComponent() {
        return <div>CssVariables</div>
    },
    AlertStack: function MockComponent() {
        return <div>AlertStack</div>
    },
    AlertBar: function MockComponent() {
        return <div>AlertBar</div>
    },
}))

useDataEngine.mockReturnValue({
    dataEngine: {},
})

useD2.mockReturnValue({
    d2: {
        currentUser: { username: 'rainbowDash' },
    },
})

const dashboards = {
    byId: {
        rainbow123: {
            id: 'rainbow123',
            displayName: 'Rainbow Dash',
            starred: false,
        },
        fluttershy123: {
            id: 'fluttershy123',
            displayName: 'Fluttershy',
            starred: true,
        },
    },
}

Object.defineProperty(global.document, 'getElementsByTagName', {
    value: () => [
        {
            classList: {
                add: Function.prototype,
                remove: Function.prototype,
            },
        },
    ],
})

Object.defineProperty(global.window, 'scrollTo', {
    value: Function.prototype,
})

describe('Dashboard', () => {
    it('renders Loading indicator when dashboards not loaded', () => {
        const store = {
            dashboards: { byId: null },
            selected: { id: null },
            user: {
                id: '135',
                username: 'test',
            },
        }

        const match = {
            params: { dashboardId: null },
        }

        apiFetchDashboard.mockReturnValue({
            id: 'rainbow123',
            displayName: 'Rainbow Dash',
            starred: false,
            dashboardItems: [],
            user: {},
            created: 'today',
            lastUpdated: 'today',
        })

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={VIEW} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })

    //it.skip('renders Redirect', () => {
    // })

    it('renders NEW dashboard', () => {
        const store = {
            dashboards,
            selected: { id: 'rainbow123' },
            user: {
                id: '135',
                username: 'test',
            },
        }

        const match = {
            params: { dashboardId: null },
        }

        apiFetchDashboard.mockReturnValue({
            id: 'rainbow123',
            displayName: 'Rainbow Dash',
            starred: false,
            dashboardItems: [],
            user: {},
            created: 'today',
            lastUpdated: 'today',
        })

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={NEW} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders No Dashboards message when no dashboards and mode != NEW', () => {
        const store = {
            dashboards: { byId: {} },
            selected: { id: null },
            user: {
                id: '135',
                username: 'test',
            },
        }

        const match = {
            params: { dashboardId: null },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={VIEW} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders Loading indicator when dashboards loaded and id is still null', () => {
        const store = {
            dashboards,
            selected: { id: null },
            user: {
                id: '135',
                username: 'test',
            },
        }

        apiFetchDashboard.mockReturnValue({
            id: 'rainbow123',
            displayName: 'Rainbow Dash',
            starred: false,
            dashboardItems: [],
            user: {},
            created: 'today',
            lastUpdated: 'today',
        })

        const match = {
            params: { dashboardId: 'rainbow123' },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={VIEW} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders Dashboard Not Found message when dashboard id is not valid', () => {
        const store = {
            dashboards,
            selected: { id: NON_EXISTING_DASHBOARD_ID },
            user: {
                id: '135',
                username: 'test',
            },
        }

        const match = {
            params: { dashboardId: 'xyzpdq' },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={VIEW} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders EDIT dashboard', () => {
        const store = {
            dashboards,
            selected: { id: 'rainbow123' },
            user: {
                id: '135',
                username: 'test',
            },
        }

        apiFetchDashboard.mockReturnValue({
            id: 'rainbow123',
            displayName: 'Rainbow Dash',
            starred: false,
            dashboardItems: [],
            user: {},
            created: 'today',
            lastUpdated: 'today',
        })

        const match = {
            params: { dashboardId: 'rainbow123' },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={EDIT} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders PRINT dashboard', () => {
        const store = {
            dashboards,
            selected: { id: 'rainbow123' },
            user: {
                id: '135',
                username: 'test',
            },
        }

        apiFetchDashboard.mockReturnValue({
            id: 'rainbow123',
            displayName: 'Rainbow Dash',
            starred: false,
            dashboardItems: [],
            user: {},
            created: 'today',
            lastUpdated: 'today',
        })

        const match = {
            params: { dashboardId: 'rainbow123' },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={PRINT} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders PRINT_LAYOUT dashboard', () => {
        const store = {
            dashboards,
            selected: { id: 'rainbow123' },
            user: {
                id: '135',
                username: 'test',
            },
        }

        apiFetchDashboard.mockReturnValue({
            id: 'rainbow123',
            displayName: 'Rainbow Dash',
            starred: false,
            dashboardItems: [],
            user: {},
            created: 'today',
            lastUpdated: 'today',
        })

        const match = {
            params: { dashboardId: 'rainbow123' },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <Router history={createMemoryHistory()}>
                        <Dashboard mode={PRINT_LAYOUT} match={match} />
                    </Router>
                </WindowDimensionsProvider>
            </Provider>
        )

        expect(container).toMatchSnapshot()
    })
})
