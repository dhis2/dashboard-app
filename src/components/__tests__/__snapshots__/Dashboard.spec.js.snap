// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders Dashboard Not Found message when dashboard id is not valid 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    DashboardsBar
  </div>
  <div>
    Requested dashboard not found
  </div>
  <div>
    AlertStack
  </div>
</div>
`;

exports[`Dashboard renders EDIT dashboard 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    EditDashboard
  </div>
  <div>
    AlertStack
  </div>
</div>
`;

exports[`Dashboard renders Loading indicator when dashboards loaded and id is still null 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    Layer
  </div>
  <div>
    AlertStack
  </div>
</div>
`;

exports[`Dashboard renders Loading indicator when dashboards not loaded 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    Layer
  </div>
  <div>
    AlertStack
  </div>
</div>
`;

exports[`Dashboard renders NEW dashboard 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    NewDashboard
  </div>
  <div>
    AlertStack
  </div>
</div>
`;

exports[`Dashboard renders No Dashboards message when no dashboards and mode != NEW 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    DashboardsBar
  </div>
  <div>
    No dashboards found. Use the + button to create a new dashboard.
  </div>
  <div>
    AlertStack
  </div>
</div>
`;

exports[`Dashboard renders PRINT dashboard 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    PrintDashboard
  </div>
  <div>
    AlertStack
  </div>
</div>
`;

exports[`Dashboard renders PRINT_LAYOUT dashboard 1`] = `
<div>
  <div>
    CssVariables
  </div>
  <div>
    PrintLayoutDashboard
  </div>
  <div>
    AlertStack
  </div>
</div>
`;
